coderd/httpmw/realip.go

```diff
@@ -0,0 +1,310 @@
+package httpmw
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/httpapi"
+)
+
+const (
+	// Note: these should be canonicalized (see http.CanonicalHeaderKey)
+	// or else things will not work correctly
+	headerCFConnectingIP  string = "Cf-Connecting-Ip"
+	headerTrueClientIP    string = "True-Client-Ip"
+	headerXRealIP         string = "X-Real-Ip"
+	headerXForwardedFor   string = "X-Forwarded-For"
+	headerXForwardedProto string = "X-Forwarded-Proto"
+)
+
+var headersAll = []string{
+	headerCFConnectingIP,
+	headerTrueClientIP,
+	headerXRealIP,
+	headerXForwardedFor,
+	headerXForwardedProto,
+}
+
+// Config configures the search order for the function, which controls
```

Comment:
impressed with all the comments throughout this!

Reply:
Jonathan wrote this originally I believe :D